### PR TITLE
Switch to mkinitrd for updating initrd img for DM-Verity.

### PR DESCRIPTION
**Pull request by liulanze**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This PR commit switches to use the default way mkinitrd to build / update the initrd img for DM-Verity, instead of using dracut to build another new initramfs img. The reasons:

1. Streamlining the update process: By updating the existing initrd, we eliminate an extra step previously required for updating initramfs.
2. Ensuring consistency during downstream operations: When the image is passed to downstream Trident for RAID operations, it necessitates the rebuilding of initrd. Adopting this approach ensures consistency in the initrd images.

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- Internal bug: 6541

###### Test Methodology

- Local build

### CODE REVIEW

**Comment by amritakohli** [comment]
Update the description - check all the boxes - you can tick boxes that don't apply to the PR as well

File: toolkit/tools/pkg/imagecustomizerlib/customizeverity.go [hunk 1]
```diff
@@ -73,59 +68,23 @@ func buildDracutModule(dracutModuleName string, imageChroot *safechroot.Chroot)
...
 		err = file.WriteLines(lines, dracutConfigFile)
 		if err != nil {
 			return fmt.Errorf("failed to write to dracut module config file (%s): %w", dracutConfigFile, err)
 		}
 	}
 
 	err = imageChroot.Run(func() error {
-		err = shell.ExecuteLiveWithErr(1, "dracut", "-f", "--kver", kernelVersion)
+		initrdImagePath := "/boot/initrd.img-" + kernelVersion
+		err = shell.ExecuteLiveWithErr(1, "mkinitrd", "-f", initrdImagePath, kernelVersion)
```
Comment (gmileka) [inline toolkit/tools/pkg/imagecustomizerlib/customizeverity.go:83]: I missed the conversation on this one, but why are we switching frim dracut to mkinitrd?

**Review by vinceaperri — APPROVED** [review]

**Review by romoh — APPROVED** [review]

File: toolkit/tools/pkg/imagecustomizerlib/customizeverity.go (hunk 1 above)
Comment (liulanze) [reply toolkit/tools/pkg/imagecustomizerlib/customizeverity.go:83]: Added to the description, please check.

**Comment by liulanze** [comment]
Pick this commit to this PR: https://github.com/microsoft/CBL-Mariner/pull/7422.

**Pull request closed without merging**